### ISSUE TITLE
DeviceInfo.getDeviceName() returns 'unknown' on Android

### ISSUE THREAD

**Issue by jcarioti** [react-native-device-info/react-native-device-info#1463]
## Summary

We just upgraded from DeviceInfo v8.3.2 to v10.1.2 and noticed that `DeviceInfo.getDeviceName()` is always returning 'unknown' on Android.

|             |     |
| ----------- | --- |
| Version     | 10.1.2   |
| Affected OS | Android   |
| OS Version  | 12, 13   |

## Current behavior

`DeviceInfo.getDeviceName()` always returns 'unknown' as the device name.

## Expected behavior

`DeviceInfo.getDeviceName()` should return the actual device name.

**Comment by mikehardy** [comment]
Hi there!
I just pulled the repo and ran the example. I can't reproduce

https://github.com/react-native-device-info/react-native-device-info/blob/7387a623a4f9a4445b81a73172fb54a9c5ef5f37/example/App.js#L207

If you can reproduce this using the example please let me know. Note that the example needs a refresh but I removed `example/yarn.lock` and re-ran `yarn example:install` to make sure it pulled fresh / updated code here from master, and it works like this

![Screenshot from 2022-09-17 13-15-46](https://user-images.githubusercontent.com/782704/190871107-90b8ab22-e558-401f-a9a3-1e1152cd212e.png)

**Comment by mikehardy** [comment]
I should mention - execution environment, I ran that test against an API33 / android 13 emulator

**Comment by jcarioti** [comment]
Okay, I finally got some time to dig into this and I realized it's due to the fact that our app is targeting API 31. That causes the `Settings.Secure.getString(getReactApplicationContext().getContentResolver(), "bluetooth_name");` call to throw an exception: `Settings key: <bluetooth_name> is only readable to apps with targetSdkVersion lower than or equal to: 31`.

That exception causes us to get thrown into the exception block so we don't even get the chance to try the `Settings.Global.getString(getReactApplicationContext().getContentResolver(), Settings.Global.DEVICE_NAME);` alternative.

My plan is to patch this for our use by wrapping the 'bluetooth_name' call in a version check. Curious if that makes the most sense to you or if there are other plans for API 31+?
